### PR TITLE
Ownable namespace manager

### PR DESCRIPTION
**Pull request by MarvinJanssen**
Still a work in progress.

### CODE REVIEW

File: backend-public/contracts/bns2.clar [hunk 1]
```diff
@@ -1,123 +1,64 @@
...
 (define-public (namespace-preorder (hashed-salted-namespace (buff 20))
                                    (stx-to-burn uint))
-    (let ((this-ctr (as-contract tx-sender)))
-        (asserts! (is-authorized tx-sender) err-not-authorized)
-        (try! (stx-transfer? stx-to-burn tx-sender this-ctr))
+    (begin
+        (try! (is-contract-owner))
+        (try! (stx-transfer? stx-to-burn tx-sender (as-contract tx-sender)))
         (to-uint-response (as-contract (contract-call? 'SP000000000000000000002Q6VF78.bns
         namespace-preorder hashed-salted-namespace stx-to-burn)))))
```
Comment (LNow) [inline backend-public/contracts/bns2.clar:12]: Have you thought about moving this call to `name-register` function? That way you'll have to submit only one TX to register the name.

Comment (friedger) [reply backend-public/contracts/bns2.clar:12]: It comes at the risk that other users front run the registration if the name is broadcast in plain text

Comment (LNow) [reply backend-public/contracts/bns2.clar:12]: Fair point.

Comment (MarvinJanssen) [reply backend-public/contracts/bns2.clar:12]: At the same time, if the owner of this contract controls registration then we could have a single function to immediately register. I think it's worth having both.

**Comment by MarvinJanssen** [comment]
@friedger do you mind if I change intent style? I have a hard time following with the brackets at the end.

File: backend-public/contracts/bns2.clar [hunk 2]
```diff
@@ -1,123 +1,64 @@
...
-        (try! (set-low-price))
-        (let ((result (contract-call? 'SP000000000000000000002Q6VF78.bns get-name-price namespace name)))
-            (try! (set-high-price))
-            result)))
+        (try! (is-contract-owner))
+        (to-bool-response (as-contract (contract-call? 'SP000000000000000000002Q6VF78.bns namespace-ready namespace)))))
 
-(define-public (name-register (namespace (buff 20))
-                              (name (buff 48))
+(define-public (name-register (name (buff 48))
```
Comment (friedger) [inline backend-public/contracts/bns2.clar:35]: @MarvinJanssen Shall we keep the same signature for `name-register`?

**Pull request merged**